File: server/src/rivulets/api/humans.py
"""Human identity directory (#14).

Read-only from here — Humans are created via POST /auth/identity (claiming
a new display name mints the row as a side effect of that claim), not a
dedicated create endpoint. This router just lists the directory so a
browser session's IdentityPicker can offer "continue as an existing
person" before claiming one.
"""

from fastapi import APIRouter
from pydantic import BaseModel
from sqlalchemy import select

from rivulets.api.deps import CurrentWorkspaceId, DbSession
from rivulets.db.models import Human

router = APIRouter(prefix="/humans", tags=["humans"])


class HumanOut(BaseModel):
    id: str
    display_name: str

    model_config = {"from_attributes": True}


@router.get("", response_model=list[HumanOut])
async def list_humans(db: DbSession, _: CurrentWorkspaceId) -> list[Human]:
    """Workspace-gated only (not CurrentHumanId) — this must be reachable
    before an identity has been claimed."""
    result = await db.execute(select(Human).order_by(Human.display_name))
    return list(result.scalars().all())

File: ui/src/lib/api/humans.ts
// Human identity directory client (#14). Read-only -- Humans are created as
// a side effect of POST /auth/identity (auth.svelte.ts's applySession),
// not through this client.

import { api } from './client';
import { auth } from './auth.svelte';

export interface Human {
	id: string;
	display_name: string;
}

export const humans = {
	list: () => api.get<Human[]>('/humans', auth.token ?? undefined)
};
